src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java

```diff
@@ -109,11 +109,18 @@ public void check(JavaMethod method, ConditionEvents events) {
                     "git4idea.GitUtil.getRepositories(com.intellij.openapi.project.Project)",
                     "git4idea.GitUtil.getRepositoryManager(com.intellij.openapi.project.Project)",
                     "git4idea.branch.GitBranchesCollection.findLocalBranch(java.lang.String)",
+                    "git4idea.branch.GitBranchesCollection.getLocalBranches()",
                     "git4idea.branch.GitBranchesCollection.findRemoteBranch(java.lang.String)",
+                    "git4idea.branch.GitBranchUtil.sortBranchNames(java.util.Collection)",
                     "git4idea.branch.GitNewBranchDialog.<init>(com.intellij.openapi.project.Project, java.util.Collection, java.lang.String, java.lang.String, boolean, boolean, boolean)",
                     "git4idea.branch.GitNewBranchDialog.showAndGetOptions()",
                     "git4idea.branch.GitNewBranchOptions.getName()",
                     "git4idea.branch.GitNewBranchOptions.shouldCheckout()",
+                    "git4idea.commands.Git.getInstance()",
+                    "git4idea.commands.Git.runCommand(git4idea.commands.GitLineHandler)",
```

Comment:
Huh should method like that be whitelisted? 🤔 this would mean they can be called on UI thread, IIRC

Reply:
Dunno about `getLocalBranches()`, whether it reads from disk or memory only